lib/src/apigateway/openapi-gateway-lambda.ts

```diff
@@ -182,11 +178,13 @@ const addLogGroupForTracing = (
 
     const deployOptionsAccessLog: StageOptions = {
       ...props.deployOptions,
-      accessLogDestination: new LogGroupLogDestination(logGroupAccessLog),
-      accessLogFormat: AccessLogFormat.jsonWithStandardFields(),
-      loggingLevel: MethodLoggingLevel.INFO,
-      dataTraceEnabled: false,
-      tracingEnabled: true,
+      accessLogDestination:
+        props.deployOptions?.accessLogDestination ?? new LogGroupLogDestination(logGroupAccessLog),
+      accessLogFormat:
+        props.deployOptions?.accessLogFormat ?? AccessLogFormat.jsonWithStandardFields(),
+      loggingLevel: props.deployOptions?.loggingLevel ?? MethodLoggingLevel.INFO,
+      dataTraceEnabled: props.deployOptions?.dataTraceEnabled ?? false,
+      tracingEnabled: props.deployOptions?.tracingEnabled ?? true,
```

Comment:
It sounds a bit weird to have `deployOptions` being controlled by the function `addLogGroupForTracing`.

@flaviostutz maybe it is ok for this PR's sake to accept the props & remove the conditional that was throwing an error, but this whole setup seems like to be in need of a revamp.

IMO `addLogGroupForTracing` should be only adding things on top a pre-existing `deployOptions` so configuring extra things on it is done outside this function's scope.

Reply:
I agree, this is confusing for me as well